Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift

```diff
@@ -1,41 +1,61 @@
 import Foundation
 import Combine
 
+import WalletConnectVerify
+
 class WalletRequestSubscriber {
     private let networkingInteractor: NetworkInteracting
     private let logger: ConsoleLogging
     private let kms: KeyManagementServiceProtocol
     private var publishers = [AnyCancellable]()
     private let walletErrorResponder: WalletErrorResponder
     private let pairingRegisterer: PairingRegisterer
-    var onRequest: ((AuthRequest) -> Void)?
-
+    private let verifyClient: VerifyClient?
+    var onRequest: (((request: AuthRequest, context: AuthContext?)) -> Void)?
+    
     init(
         networkingInteractor: NetworkInteracting,
         logger: ConsoleLogging,
         kms: KeyManagementServiceProtocol,
         walletErrorResponder: WalletErrorResponder,
-        pairingRegisterer: PairingRegisterer) {
+        pairingRegisterer: PairingRegisterer,
+        verifyClient: VerifyClient?
+    ) {
         self.networkingInteractor = networkingInteractor
         self.logger = logger
         self.kms = kms
         self.walletErrorResponder = walletErrorResponder
         self.pairingRegisterer = pairingRegisterer
+        self.verifyClient = verifyClient
         subscribeForRequest()
     }
-
+    
     private func subscribeForRequest() {
         pairingRegisterer.register(method: AuthRequestProtocolMethod())
             .sink { [unowned self] (payload: RequestSubscriptionPayload<AuthRequestParams>) in
                 logger.debug("WalletRequestSubscriber: Received request")
-
+                
                 pairingRegisterer.activate(
                     pairingTopic: payload.topic,
                     peerMetadata: payload.request.requester.metadata
                 )
-
+                
                 let request = AuthRequest(id: payload.id, payload: payload.request.payloadParams)
-                onRequest?(request)
+                
+                if let rawRequest = payload.rawRequest, let verifyClient {
```

Comment:
I see the same validation code it SessionEngine, is it possible to do it in one-two lines? May be hide it in some kind of ValidationService and reuse